### ISSUE TITLE
The index overflow when running function 'multinomial'

### ISSUE THREAD

**Issue by liuxz-cs** [pytorch/pytorch#44866]
### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

When I use the function multinomial to get the result, the result overflows, and get the value 3223520557, I want to know how can I solve the problem. 

I use the code `action = prob.multinomial(1).data` to get the index of action, and the value of prob is `prob = tensor([[1.3229, 0.0526, 0.0467, 0.0164, 0.2433, 0.1570]], device='cuda:0', grad_fn=<SoftmaxBackward>) ` but the value of action is `action = tensor([[3223520557]], device='cuda:0')`

I did not get the problem before, and I have got this error recently.  So what's the probable errors?

cc @vincentqb @fritzo @neerajprad @alicanb @vishwakftw @nikitaved

**Comment by glaringlee** [comment]
@liuxz-cs 
I tried to reproduce this on my end, but I can not get 3223520557, I got 0 instead. (I create the `prob`  by torch.tensor, not calling softmax).

Can you provide more info about your code? 
a. If you are using an older version pytorch, can you try building the latest pytorch using the code in master branch?
b. it seems to me that `prob` comes from softmax. Can you provide a code piece which I can reproduce the error?
Thanks.

**Comment by liuxz-cs** [comment]
> @liuxz-cs
> I tried to reproduce this on my end, but I can not get 3223520557, I got 0 instead. (I create the `prob` by torch.tensor, not calling softmax).
> 
> Can you provide more info about your code?
> a. If you are using an older version pytorch, can you try building the latest pytorch using the code in master branch?
> b. it seems to me that `prob` comes from softmax. Can you provide a code piece which I can reproduce the error?
> Thanks.

The piece of code I use is 
`
prob = F.softmax(out.logit, dim=1)
action = prob.multinomial(1).data
`
actually, out.logit is the output of a CNN network, it is a vector with shape [1 x 6].

The torch version I use is 0.4.1.post2, and the cuda version is 9.0 the cudnn version is 7.0.5
I do not know if the version of torch, cuda and cudnn not match leading to that error.

**Comment by glaringlee** [comment]
@liuxz-cs 
0.4.1.post2 is too old. Can you install a newer version and try again?
You can find the installation instruction at https://pytorch.org/
In 'QUICK START LOCALLY' session.

**Comment by liuxz-cs** [comment]
> @liuxz-cs
> 0.4.1.post2 is too old. Can you install a newer version and try again?
> You can find the installation instruction at https://pytorch.org/
> In 'QUICK START LOCALLY' session.

So what version should I install to math my cuda and cudnn version?

**Comment by liuxz-cs** [comment]
> > @liuxz-cs
> > 0.4.1.post2 is too old. Can you install a newer version and try again?
> > You can find the installation instruction at https://pytorch.org/
> > In 'QUICK START LOCALLY' session.
> 
> So what version should I install to math my cuda and cudnn version?

the OS is ubuntu 16.04

**Comment by glaringlee** [comment]
@liuxz-cs 
Can you install cuda 9.2? Our newer version pytorch support cuda from version 9.2, if you can install 9.2, then you can install the latest version of pytorch.

If you want to stick with 9.0. Try pytorch 1.1. 
You can find the installation info here:
https://pytorch.org/get-started/previous-versions/
Search version 1.1

**Comment by liuxz-cs** [comment]
> @liuxz-cs
> Can you install cuda 9.2? Our newer version pytorch support cuda from version 9.2, if you can install 9.2, then you can install the latest version of pytorch.
> 
> If you want to stick with 9.0. Try pytorch 1.1.
> You can find the installation info here:
> https://pytorch.org/get-started/previous-versions/
> Search version 1.1

Okay, I will try it, thanks a lot!

**Comment by ngimel** [comment]
Closing, please reopen if this reproduces on the latest version.
